### PR TITLE
PE-28 | Created profile onboarding.

### PR DESCRIPTION
**Pull request by tonynguyen98**
**What I did:** 

- Created a page that allows new users to input their information. We can also reuse this page for updating the profile. 
- Created a blank page to redirect to upon valid submission. 

**How to test:** 

- Test each field to when blank to see if the validator is working. It should NOT redirect to a new page if invalid. 
- Fill in your information as normal and test to see if it will submit and redirect properly when everything is correct. Also make sure that in the terminal, there is a printout of the values you've inputted.

### CODE REVIEW

**Comment by ayeejane** [comment]
Code looks good! Only issue is when entering fields for First Name, Last Name, and City, there is an overflow of pixels and it shows when the keyboard pops up.

**Review by ayeejane — CHANGES_REQUESTED** [review]
Code looks good! Only issue is when entering fields for First Name, Last Name, and City, there is an overflow of pixels and it shows when the keyboard pops up.

File: lib/profile-onboarding.dart [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+    Function onSaved,
+  ) {
+    return Column(
+      children: <Widget>[
+        TextFormField(
+          textCapitalization: TextCapitalization.sentences,
+          keyboardType: TextInputType.text,
+          textInputAction: textInputAction,
+          validator: (value) {
+            if (value.isEmpty) {
```
Comment (MaxLian11) [inline lib/profile-onboarding.dart]: I would also add restrictions: 
- strings consisting of only spaces
- strings not containing letters
- strings containing numbers

File: lib/profile-onboarding.dart [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+      ),
+    ),
+  };
+
+  Column segmentedControl() {
+    return Column(
+      children: [
+        Container(
+          padding: EdgeInsets.fromLTRB(0.0, 0.0, 0.0, 10.0),
+          child: Text('Do you have a preference?'),
```
Comment (MaxLian11) [inline lib/profile-onboarding.dart]: I would change this question a little bit. Is it possible to make a question that would explain the user what this means a how it affects the functionality?

File: lib/profile-onboarding.dart [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+  }
+
+  DateTimePicker birthDateSelector() {
+    return DateTimePicker(
+      initialValue: '',
+      firstDate: DateTime(1930),
+      lastDate: DateTime(2030),
+      decoration: InputDecoration(
+        border: OutlineInputBorder(),
+        labelText: 'Birthday',
```
Comment (MaxLian11) [inline lib/profile-onboarding.dart]: "Date of Birth" sounds better to me

**Review by MaxLian11 — CHANGES_REQUESTED** [review]
All in all it looks very good! I have the same issue as Jane - when typing, it shows overflow of pixels, not sure what causes this issue. I am using Nexus 6 API 30 (1440 x 2560px)

**Review by dominikyz — CHANGES_REQUESTED** [review]
Code looks great overall.
I'm getting pixel overflow error when entering first name, last name, and city fields.
As Max suggested, change birthday to Date of Birth. ( Do we really need date of birth? i think we don't need this field at all, since app is not age restrictive )
If its not hard to implement, adding feature to close keyboard on clicking anywhere on screen would be nice.

**Comment by tonynguyen98** [comment]
> As Max suggested, change birthday to Date of Birth. ( Do we really need date of birth? i think we don't need this field at all, since app is not age restrictive )

While we're not really using it for anything, it probably doesn't hurt to implement this right since it's only a couple extra lines of code.

File: lib/profile-onboarding.dart (hunk 1 above)
Comment (tonynguyen98) [reply lib/profile-onboarding.dart]: While I agree there should be some restrictions, it'll be very hard to find the proper regex for that won't limit certain people. Some people might be named "de Rune" or "X Æ A-12"

**Review by MaxLian11 — APPROVED** [review]
Looks good! Please, add more comments and notes next time

**Review by dominikyz — APPROVED** [review]
Code compiles without errors now :) Great job!

File: lib/profile-onboarding.dart [hunk 4]
```diff
@@ -96,58 +109,88 @@ class _ProfileOnboardingState extends State<ProfileOnboarding> {
...
-  DateTimePicker birthDateSelector() {
+  DateTimePicker _birthDateSelector() {
     return DateTimePicker(
       initialValue: '',
       firstDate: DateTime(1930),
       lastDate: DateTime(2030),
       decoration: InputDecoration(
         border: OutlineInputBorder(),
-        labelText: 'Birthday',
+        labelText: 'Date of Birth',
```
Comment (ayeejane) [inline lib/profile-onboarding.dart]: The calendar popping up and having to either scroll up for your birthdate or editing it seems unnecessary. I would just have the user enter in their birthdate in the form: 01/20/2020.

**Review by ayeejane — CHANGES_REQUESTED** [review]

File: lib/profile-onboarding.dart (hunk 4 above)
Comment (ayeejane) [reply lib/profile-onboarding.dart]: Other than that, everything else works and looks good!

File: lib/profile-onboarding.dart [hunk 5]
```diff
@@ -0,0 +1,272 @@
...
+            labelText: labelText,
+          ),
+          onSaved: onSaved,
+        ),
+        _sizedBoxPadding(),
+      ],
+    );
+  }
+
+  Row _birthDateForms() {
```
Comment (ayeejane) [inline lib/profile-onboarding.dart]: Looks good to me! Great change on the birthday input.

**Review by ayeejane — APPROVED** [review]

**Pull request merged**